runtime/bcverify/staticverify.c

```diff
@@ -228,26 +228,58 @@ checkBytecodeStructure (J9CfrClassFile * classfile, UDATA methodIndex, UDATA len
 				errorDataIndex = index;
 				goto _verifyError;
 			}
+
 			info = &(classfile->constantPool[index]);
 			tag = (UDATA) info->tag;
-
-			if (!((tag == CFR_CONSTANT_Integer)
-				  || (tag == CFR_CONSTANT_Float)
-				  || (tag == CFR_CONSTANT_String)
-				  || ((tag == CFR_CONSTANT_Class)
-							&& (((flags & BCT_MajorClassFileVersionMask) >= BCT_Java5MajorVersionShifted)
-									|| ((flags & BCT_MajorClassFileVersionMask) == 0)))
-				  || (((tag == CFR_CONSTANT_MethodType) || (tag == CFR_CONSTANT_MethodHandle))
-						  && (((flags & BCT_MajorClassFileVersionMask) >= BCT_Java7MajorVersionShifted)
-									|| ((flags & BCT_MajorClassFileVersionMask) == 0)))	)) {
-				errorType = J9NLS_CFR_ERR_BC_LDC_NOT_CONSTANT__ID;
-				/* Jazz 82615: Set the constant pool index to show up in the error message framework */
-				errorDataIndex = index;
-				goto _verifyError;
+			{
+				switch (tag) {
+				case CFR_CONSTANT_Integer:
+				case CFR_CONSTANT_Float:
+				case CFR_CONSTANT_String:
+					break;
+				case CFR_CONSTANT_Class:
+					if (((flags & BCT_MajorClassFileVersionMask) < BCT_Java5MajorVersionShifted)
+					&& (J9_ARE_ANY_BITS_SET(flags, BCT_MajorClassFileVersionMask))
```

Comment:
given the way this has been rewritten, it makes sense to just remove the `&& (J9_ARE_ANY_BITS_SET(flags, BCT_MajorClassFileVersionMask))` as I don't believe it's required anymore.

Previously, cfdump used it.